src/main/java/de/tum/in/www1/artemis/service/GroupNotificationService.java

```diff
@@ -7,220 +7,145 @@
+     * Notify student groups about an exercise update.
      *
      * @param exercise that has been updated
-     * @param notificationText that should be displayed
+     * @param notificationText that should be displayed  TODO: can be removed in the future as notification's text attribute is not used in the client
      */
     public void notifyStudentGroupAboutExerciseUpdate(Exercise exercise, String notificationText) {
+        // Do not send a notification before the release date of the exercise.
         if (exercise.getReleaseDate() != null && exercise.getReleaseDate().isAfter(ZonedDateTime.now())) {
             return;
         }
-        String title = "Exercise \"" + exercise.getTitle() + "\" updated";
-        notifyStudentGroupAboutExerciseChange(exercise, title, notificationText);
-    }
-
-    public void notifyInstructorGroupAboutExerciseUpdate(Exercise exercise, String notificationText) {
-        String title = "Exercise \"" + exercise.getTitle() + "\" updated";
-        notifyInstructorGroupAboutExerciseChange(exercise, title, notificationText);
-    }
-
-    private void notifyInstructorGroupAboutExerciseChange(Exercise exercise, String title, String notificationText) {
-        GroupNotification groupNotification = createExerciseUpdatedGroupNotificationForInstructors(exercise, title, notificationText);
-        saveAndSendGroupNotification(groupNotification);
-    }
-
-    private void notifyStudentGroupAboutExerciseChange(Exercise exercise, String title, String notificationText) {
-        GroupNotification groupNotification = createExerciseUpdatedGroupNotificationForStudents(exercise, title, notificationText);
-        saveAndSendGroupNotification(groupNotification);
+        // Create and send the notification.
+        saveAndSend(GroupNotificationFactory.createNotification(exercise, userService.getUser(), GroupNotificationType.STUDENT, NotificationType.EXERCISE_UPDATED));
     }
 
     /**
-     * Notify tutor group about the creation of an exercise
+     * Notify tutor groups about the creation of an exercise.
      *
      * @param exercise that has been created
      */
     public void notifyTutorGroupAboutExerciseCreated(Exercise exercise) {
-        GroupNotification groupNotification = createExerciseCreatedGroupNotificationForTutors(exercise);
-        saveAndSendGroupNotification(groupNotification);
+        saveAndSend(GroupNotificationFactory.createNotification(exercise, userService.getUser(), GroupNotificationType.TA, NotificationType.EXERCISE_CREATED));
+    }
+
+    /**
+     * Notify instructor groups about an exercise update.
+     *
+     * @param exercise that has been updated
+     * @param notificationText that should be displayed  TODO: can be removed in the future as notification's text attribute is not used in the client
+     */
+    public void notifyInstructorGroupAboutExerciseUpdate(Exercise exercise, String notificationText) {
+        // TODO: why null for author?
```

Comment:
Who is this question directed at? Maybe it'd be better to clarify the question since the `TODO` adds more confusion than it alleviates, I feel.

Reply:
I forgot this and the previous TODO. Was somehow a reminder for myself. I will resolve both todos.